pkgs/tools/networking/linux-router/default.nix

```diff
@@ -0,0 +1,102 @@
+# SOURCE: https://github.com/NixOS/nixpkgs/pull/92787/files
+
+{ stdenv, lib, pkgs, fetchFromGitHub
+
+# Dependencies
+, bash , procps , iproute2 , dnsmasq , iptables
+
+# WiFi hotspot dependencies
+, hostapd , iw
+, wirelesstools ? null # for iwconfig
+, haveged ? null
+, qrencode ? null
+
+# General dependencies
+, coreutils , flock , gawk , getopt , gnugrep , gnused , which
+, networkmanager ? null
+
+# To fall back to haveged if entropy is low.
+#
+# Defaulting to false because not having it does not break things.
+# If it is really needed, warnings will be logged to journal.
+, useHaveged ? false
+
+# nmcli is not required for create_ap.
+# (defaulting to true because it is very likely already present)
+, useNetworkManager ? true
+
+# You only need this if 'iw' can not recognize your adapter.
+, useWirelessTools ? true
+
+# You only need this if you wish to show WiFi QR codes in terminal
+, useQrencode ? true
+}:
+
+assert useHaveged        -> !isNull haveged;
+assert useNetworkManager -> !isNull networkmanager;
+assert useWirelessTools  -> !isNull wirelesstools;
+assert useQrencode       -> !isNull qrencode;
+
+with lib;
+
+let
+  binPath = makeBinPath ([ iw procps hostapd iproute2 getopt bash dnsmasq
+    iptables coreutils which flock gnugrep gnused gawk ]
+    ++ optional useHaveged        haveged
+    ++ optional useNetworkManager networkmanager
+    ++ optional useWirelessTools  wirelesstools
+    ++ optional useQrencode       qrencode);
+
+in
+
+stdenv.mkDerivation rec {
+  pname = "linux-router";
+  version = "0.6.2";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "garywill";
+    repo = "linux-router";
+    rev = "${version}";
+    sha256 = "193bnlwmjxsk0cri6xdylf218qayldn02pdnppvbd39ls361776z";
+  };
+
+  nativeBuildInputs = [ pkgs.makeWrapper ];
+
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/bin/ $out/.bin-wrapped
+    mv lnxrouter $out/.bin-wrapped/lnxrouter
+    makeWrapper $out/.bin-wrapped/lnxrouter $out/bin/lnxrouter --prefix PATH : ${binPath}
+  '';
+
+  meta = let
+      homepage = https://github.com/garywill/linux-router;
+    in
+    {
+      inherit homepage;
+      description = "Set Linux as router in one command. Support Internet sharing, redsocks, Wifi hotspot, IPv6. Can also be used for routing VM/containers";
```

Comment:
```suggestion
      description = "Set Linux as router in one command";
```

Reply:
Why would you remove that? Is there a recommended max length for `description`? I personally think that it gives a much better summary for what this tool does than just saying "Set Linux as router" which is - imho - too generic/abstract.

Reply:
I think it is too long. Package descriptions are typically not more than one sentence (https://nixos.org/manual/nixpkgs/stable/#var-meta-description). You also have the long description which lists all these features.

Reply:
How about: "Set Linux as router/Wifi hotspot/proxy in one command"

Reply:
I'd be ok with that but I'd like to point out that the description I used was straight from the projects GitHub repo (the short description at the top right)